python/ray/data/tests/test_block_batching.py

```diff
@@ -121,6 +123,96 @@ def test_format_batches(batch_format):
             assert isinstance(batch["foo"], np.ndarray)
 
 
+def test_async_batch_fetching():
```

Comment:
shall we add a test for `map_batches` as well?

Reply:
I tried it but there's too much time variance for a deterministic small-scale map_batches CI test. I'll confirm the performance improvements via running the batch inference release tests.